src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Tree/Dependencies/Models/DependencyModel.cs

```diff
@@ -19,21 +19,23 @@ private enum DependencyFlags : byte
         }
 
         protected DependencyModel(
-            string path,
+            string? path,
             string originalItemSpec,
             ProjectTreeFlags flags,
             bool isResolved,
             bool isImplicit,
             IImmutableDictionary<string, string>? properties,
             bool isVisible = true)
         {
-            Requires.NotNullOrEmpty(path, nameof(path));
+            // IDependencyModel allows original item spec to be null, but we can satisfy a
+            // more strict requirement for the dependency types produced internally.
+            // External providers may not have a meaningful value, but do not use this type.
             Requires.NotNullOrEmpty(originalItemSpec, nameof(originalItemSpec));
 
             Path = path;
             OriginalItemSpec = originalItemSpec;
             Properties = properties ?? ImmutableStringDictionary<string>.EmptyOrdinal;
-            Caption = path;
+            Caption = originalItemSpec;
```

Comment:
This looks strange that we would use original item spec without any manipulation for items, what items does this affect?

Reply:
This is the base class ctor. Most subtypes reassign `Caption` with type-specific logic that's often too involved to do inline in the `base(...)` statement.